service/collector/cluster.go

```diff
@@ -90,7 +92,9 @@ func (c *Cluster) Collect(ch chan<- prometheus.Metric) error {
 				key.ObjRefToNamespacedName(key.ObjRefFromCluster(cl)),
 				cr,
 			)
-			if err != nil {
+			if apierrors.IsNotFound(err) {
+				c.logger.LogCtx(ctx, "level", "warning", "message", fmt.Sprintf("could not find object reference %#q", cl.GetName()))
```

Comment:
I think we need to `continue` here. Falling through will just generate garbage metrics.